editor/src/components/canvas/canvas-strategies/shared-absolute-move-strategy-helpers.ts

```diff
@@ -30,22 +23,130 @@ import {
   withUnderlyingTarget,
 } from '../../editor/store/editor-state'
 import { stylePropPathMappingFn } from '../../inspector/common/property-path-hooks'
-import { CanvasFrameAndTarget } from '../canvas-types'
+import { determineConstrainedDragAxis } from '../canvas-controls-frame'
+import { CanvasFrameAndTarget, CSSCursor } from '../canvas-types'
 import {
   adjustCssLengthProperty,
   AdjustCssLengthProperty,
 } from '../commands/adjust-css-length-command'
-import { runLegacyAbsoluteMoveSnapping } from '../controls/guideline-helpers'
+import { CanvasCommand } from '../commands/commands'
+import { pushIntendedBounds } from '../commands/push-intended-bounds-command'
+import { setCursorCommand } from '../commands/set-cursor-command'
+import { setElementsToRerenderCommand } from '../commands/set-elements-to-rerender-command'
+import { setSnappingGuidelines } from '../commands/set-snapping-guidelines-command'
+import { updateHighlightedViews } from '../commands/update-highlighted-views-command'
+import {
+  collectParentAndSiblingGuidelines,
+  runLegacyAbsoluteMoveSnapping,
+} from '../controls/guideline-helpers'
 import {
   ConstrainedDragAxis,
-  Guideline,
   GuidelineWithRelevantPoints,
   GuidelineWithSnappingVectorAndPointsOfRelevance,
 } from '../guideline'
 import { AbsolutePin } from './absolute-resize-helpers'
-import { InteractionCanvasState } from './canvas-strategy-types'
+import {
+  emptyStrategyApplicationResult,
+  getTargetPathsFromInteractionTarget,
+  InteractionCanvasState,
+  StrategyApplicationResult,
+  strategyApplicationResult,
+} from './canvas-strategy-types'
 import { InteractionSession, StrategyState } from './interaction-state'
 
+export const getAdjustMoveCommands =
+  (
+    canvasState: InteractionCanvasState,
+    interactionState: InteractionSession,
+    sessionState: StrategyState,
+  ) =>
+  (
+    snappedDragVector: CanvasPoint,
+  ): {
+    commands: Array<AdjustCssLengthProperty>
+    intendedBounds: Array<CanvasFrameAndTarget>
+  } => {
+    const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+    const filteredSelectedElements = getDragTargets(selectedElements)
+    let commands: Array<AdjustCssLengthProperty> = []
+    let intendedBounds: Array<CanvasFrameAndTarget> = []
+    filteredSelectedElements.forEach((selectedElement) => {
+      const elementResult = getAbsoluteMoveCommandsForSelectedElement(
```

Comment:
This function name is making me raise my eyebrow given that `getAdjustMoveCommands` is being used by the relative positioning strategy. I took a quick look at the function itself and it looks like it is still fine to use for relative positioning, in which case please can we rename it. Also, these functions should be moved to a more generically named file since they're now being used by non-absolute strategies.

Reply:
yeah good points, will do

Reply:
https://github.com/concrete-utopia/utopia/pull/2604/commits/b355cdb0a31eadada31eda5ae4723ce613f2aa00 note: aside from renaming the mentioned function, I just renamed the whole helpers file to `shared-move-strategies-helpers` as imho it makes sense to not split it into two (e.g. one for the relative and one for the absolute movements) if one just includes both